test/multiple-user-principal-types.test.js

```diff
@@ -0,0 +1,978 @@
+
+    it('Logout a user from user model 1 without logout user from model 2', function(done) {
+      User.login({email: 'foo@bar.com', password: 'bar'}, function(err, accessToken1) {
+        User.login({email: 'foo@bar.com', password: 'bar'}, function(err, accessToken2) {
+          User.logout(accessToken1.id, function(err, res) {
+            AccessToken.findById(accessToken1.id, function(err, accessToken) {
+              assert(!accessToken, 'accessToken should not exist after logging out user1');
+              AccessToken.findById(accessToken2.id, function(err, accessToken2) {
+                assert(accessToken2, 'accessToken2 should exist after logging out user1');
+                done();
+              });
+            });
+          });
+        });
+      });
+    });
+  });
+
+  describe('Password Reset', function() {
+    describe('User.resetPassword(options, cb)', function() {
+      var options = {
+        email: 'foo@bar.com',
+        redirect: 'http://foobar.com/reset-password',
+      };
+
+      it('Creates a temp accessToken to allow a user to change password', function(done) {
+        var calledBack = false;
+
+        User.resetPassword({
+          email: options.email,
+        }, function() {
+          calledBack = true;
+        });
+
+        User.once('resetPasswordRequest', function(info) {
+          assert(info.email);
+          assert(info.accessToken);
+          assert(info.accessToken.id);
+          assert.equal(info.accessToken.ttl / 60, 15);
+          assert(calledBack);
+          info.accessToken.user(function(err, user) {
+            if (err) return done(err);
+            assert.equal(user.email, options.email);
+
+            done();
+          });
+        });
+      });
+
+      describe('User.resetPassword(options, cb) requiring realm', function() {
+        var realmUser;
+
+        beforeEach(function(done) {
+          User.create(validCredentialsWithRealm, function(err, u) {
+            if (err) return done(err);
+
+            realmUser = u;
+            done();
+          });
+        });
+
+        it('Creates a temp accessToken to allow user in realm to change password', function(done) {
+          var calledBack = false;
+
+          User.resetPassword({
+            email: realmUser.email,
+            realm: realmUser.realm,
+          }, function() {
+            calledBack = true;
+          });
+
+          User.once('resetPasswordRequest', function(info) {
+            assert(info.email);
+            assert(info.accessToken);
+            assert(info.accessToken.id);
+            assert.equal(info.accessToken.ttl / 60, 15);
+            assert(calledBack);
+            info.accessToken.user(function(err, user) {
+              if (err) return done(err);
+
+              assert.equal(user.email, realmUser.email);
+
+              done();
+            });
+          });
+        });
+      });
+    });
+  });
+
+  describe('AccessToken (session) invalidation', function() {
```

Comment:
done, well, it bubbled up a missing part in my code coverage: 
invalidating a user from model1 invalidated a user with same userId from model2 because of the missing principalType in the AccessToken.deleteAll call.
Would have missed it if i did not remove the guid function for id generation which have made them random. We shall never put this guid function at work in this file